File: core-examples/src/main/rb/io/vertx/example/core/verticle/asyncstart/deploy_example.rb

puts "Main verticle has started, let's deploy some others..."

# Deploy another instance and  want for it to start
$vertx.deploy_verticle("io.vertx.example.core.verticle.asyncstart.OtherVerticle") { |res,res_err|
  if (res_err == nil)

    deploymentID = res

    puts "Other verticle deployed ok, deploymentID = #{deploymentID}"

    $vertx.undeploy(deploymentID) { |res2,res2_err|
      if (res2_err == nil)
        puts "Undeployed ok!"
      else
        res2_err.print_stack_trace()
      end
    }
  else
    res_err.print_stack_trace()
  end
}

File: core-examples/src/main/js/io/vertx/example/core/verticle/asyncstart/deploy_example.js

console.log("Main verticle has started, let's deploy some others...");

// Deploy another instance and  want for it to start
vertx.deployVerticle("io.vertx.example.core.verticle.asyncstart.OtherVerticle", function (res, res_err) {
  if (res_err == null) {

    var deploymentID = res;

    console.log("Other verticle deployed ok, deploymentID = " + deploymentID);

    vertx.undeploy(deploymentID, function (res2, res2_err) {
      if (res2_err == null) {
        console.log("Undeployed ok!");
      } else {
        res2_err.printStackTrace();
      }
    });
  } else {
    res_err.printStackTrace();
  }
});


